File: cpp/model/SimpleEnemy.cpp
#include "SimpleEnemy.h"

SimpleEnemy::SimpleEnemy(Armor* armor, vector<Buff*> &buffs) : armor(armor), buffs(buffs) {

}

const bool SimpleEnemy::isPlayer() {
    return false;
}

const Armor* SimpleEnemy::getArmor() {
    return armor;
}

const vector<Buff*> SimpleEnemy::getBuffs() {
    return buffs;
}

File: csharp/FantasyBattle/SimpleEnemy.cs
using System.Collections.Generic;

namespace FantasyBattle
{
    public class SimpleEnemy : Target
    {
        public virtual Armor Armor { get; }
        public virtual List<Buff> Buffs { get; }

        public SimpleEnemy()
        {

        }

        public SimpleEnemy(Armor armor, List<Buff> buffs)
        {
            Armor = armor;
            Buffs = buffs;
        }
    }

    public interface Buff
    {
        float SoakModifier { get; }
        float DamageModifier { get; }
    }

    public interface Armor
    {
        int DamageSoak { get; }
    }
}